proto/sql_api.proto

```diff
@@ -0,0 +1,103 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Vivek Menezes (vivek@cockroachlabs.com) 
+
+syntax = "proto2";
+package cockroach.proto;
+option go_package = "proto";
+
+import "cockroach/proto/api.proto";
+import "cockroach/proto/data.proto";
+import "cockroach/proto/errors.proto";
+import "gogoproto/gogo.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// CmdRequestHeader is supplied with every CmdRequest.
+message CmdRequestHeader {
+  // Session settings that were returned in the last response that
+  // contained them, being reflected back to the server.
+  optional bytes session = 1;
+  // The transaction state returned in the previous response being
+  // reflected back.
+  optional bytes txn = 2;
+  // CmdID is optionally specified for request idempotence
+  // (i.e. replay protection).
+  optional ClientCmdID cmd_id = 3 [(gogoproto.nullable) = false, (gogoproto.customname) = "CmdID"];
+  // User is the originating user. Used to lookup priority when
+  // scheduling queued operations at target node.
+  optional string user = 4 [(gogoproto.nullable) = false];
+}
+
+// CmdResponseHeader is returned with every Cmd response.
+message CmdResponseHeader {
+  // Error is non-nil if an error occurred.
+  optional Error error = 1;
+  // Setting that should be reflected back in all subsequent requests.
+  // When not set, future requests should continue to use existing settings.
+  optional bytes settings = 2;
+  // Transaction message returned in a response; to be not interpreted by
+  // the sender and reflected in a subsequent request. When not set,
+  // the subsequent request should not contain a transaction object.
+  optional bytes txn = 3;
+}
+
+message Datum {
+  option (gogoproto.onlyone) = true;
+  oneof value {
+    bool bval = 1;
+    int64 ival = 2;
+    double dval = 3;
+    string sval = 4;
+    bytes blobval = 5;
```

Comment:
Oh, and to answer your question about a lastname, you would either use a `CHAR`, `VARCHAR` or `TEXT` column, but the data would be passed on the wire as `bytes`.